### PR TITLE
Fix performance regression in single topology conformation optimization

### PR DESCRIPTION
**Pull request by mcwitt**
#959 inadvertently changed the behavior of `setup_initial_states` to rerun host minimization for each lambda window. Host optimization is independent of lambda and should be run only once.

This regression was breaking nightly tests on master:

- Last good run on master (2h 41m): https://gitlab.com/relaytx/physics-platform/timemachine/-/pipelines/793714466
- Failed run on current master (timed out at 6h): https://gitlab.com/relaytx/physics-platform/timemachine/-/pipelines/794987570
- Successful run on this branch (2h 51m): https://gitlab.com/relaytx/physics-platform/timemachine/-/pipelines/795620590

The time is increased by ~10 minutes relative to last passing, but this is expected since we run RBFE tests with bisection in addition to sequentially as of #959.

### CODE REVIEW

File: timemachine/fe/rbfe.py [hunk 1]
```diff
@@ -174,8 +188,8 @@ def setup_initial_states(
     st: SingleTopology
         A single topology object
 
-    host_config: HostConfig or None
-        Configurations of the host. If None, then a vacuum state will be setup.
```
Comment (jkausrelay) [inline timemachine/fe/rbfe.py:178]: nit: Use `setup_optimized_host` to generate the host.

**Review by jkausrelay — APPROVED** [review]

File: timemachine/fe/rbfe.py (hunk 1 above)
Comment (mcwitt) [reply timemachine/fe/rbfe.py:178]: Clarified in https://github.com/proteneer/timemachine/pull/966/commits/2ea7cabb92785738d8e6199b94d1f8545f271a5e

File: timemachine/fe/rbfe.py [hunk 2]
```diff
@@ -98,10 +99,18 @@ def assert_all_states_have_same_masses(initial_states: List[InitialState]):
     np.testing.assert_array_almost_equal(deviation_among_windows, 0, err_msg="masses assumed constant w.r.t. lambda")
 
 
+@dataclass
+class Host:
+    system: VacuumSystem
+    masses: List[float]
```
Comment (maxentile) [inline timemachine/fe/rbfe.py:105]: [nonblocking, unrelated to fixing test]

Possibly rename to `physical_masses`, or have 2 fields `physical_masses` and `hmr_masses`? (Current usages look correct, but good to preempt accidentally applying HMR 0 or 2 times.)

**Review by maxentile — APPROVED** [review]
Good catch -- fix looks good!

File: timemachine/fe/rbfe.py (hunk 2 above)
Comment (mcwitt) [reply timemachine/fe/rbfe.py:105]: Ah, I thought we currently disambiguated by prefixing HMR masses with `hmr_`. (I'm not immediately seeing where this field is used for HMR masses). Is the idea just to be extra explicit?

Comment (maxentile) [reply timemachine/fe/rbfe.py:105]: Sorry, this is tangential -- will double-check this separately!

>Ah, I thought we currently disambiguated by prefixing HMR masses with hmr_. (I'm not immediately seeing where this field is used for HMR masses)

Good point -- the current convention `masses` vs `hmr_masses` is unambiguous already!

Comment (mcwitt) [reply timemachine/fe/rbfe.py]: Renamed in https://github.com/proteneer/timemachine/pull/966/commits/271007e6c96ba80734743eda7d2a96e1d1309bfe

**Pull request merged**